python-checks/src/main/resources/org/sonar/l10n/py/rules/python/S1523.html

```diff
@@ -0,0 +1,69 @@
+  <li> the code to run is dynamically chosen via reflection, and an untrusted source can use it to choose which code to run. For example a class could
+  be retrieved by its name and this name comes from a user input. </li>
+</ul>
+<p>You are at risk if you answered yes to any of these questions.</p>
+<h2>Recommended Secure Coding Practices</h2>
+<p>Regarding the execution of unknown code, the best solution is to not run code provided by an untrusted source. If you really need to do it, run the
+code in a <a href="https://en.wikipedia.org/wiki/Sandbox_(computer_security)">sandboxed</a> environment. Use jails, firewalls and whatever means your
+operating system and programming language provide (example: <a
+href="https://wiki.sei.cmu.edu/confluence/display/java/SEC54-J.+Create+a+secure+sandbox+using+a+security+manager">Security Managers</a> in java, <a
+href="https://www.w3schools.com/tags/att_iframe_sandbox.asp">iframes</a> and <a href="https://en.wikipedia.org/wiki/Same-origin_policy">same-origin
+policy</a> for javascript in a web browser).</p>
+<p>Do not try to create a blacklist of dangerous code. It is impossible to cover all attacks that way.</p>
+<p>As for the use of reflection, it should be strictly controlled as it can lead to many vulnerabilities. Never let an untrusted source decide what
+code to run. If you have to do it anyway, create a list of allowed code and choose among this list.</p>
+<h2>Exceptions</h2>
+<p>Calling reflection methods with a hard-coded type name, method name or field name will not raise an issue.</p>
+<h2>See</h2>
+<ul>
+  <li> <a href="http://cwe.mitre.org/data/definitions/95.html">MITRE CWE-95</a> - Improper Neutralization of Directives in Dynamically Evaluated Code
+  ('Eval Injection') </li>
+  <li> <a href="http://cwe.mitre.org/data/definitions/470.html">MITRE CWE-470</a> - Use of Externally-Controlled Input to Select Classes or Code
+  ('Unsafe Reflection') </li>
+  <li> OWASP Top 10 2017 Category A1 - Injection </li>
+  <li> OWASP Top 10 2017 Category A7 - Cross-Site Scripting (XSS) </li>
+</ul>
+<h2>Questionable Code Example</h2>
+<pre>
+import os
+
+value = input()
+hardcoded_command = 'os.system("ls")'
+formatted_command = 'os.system("%s")' % value
+compiled_command = compile(hardcoded_command, '&lt;string&gt;', 'eval')
+compiled_code = compile(hardcoded_command, '&lt;string&gt;', 'exec')
+
+def evaluate(command, file, mode):
+    eval(command)  # Questionable. Dynamic command
+    eval(compile(command, file, mode))  # Questionable. Dynamic command
+
+eval(hardcoded_command)  # Ok
+eval(formatted_command)  # Questionable. Dynamic code
+eval(compiled_command)  # Questionable even though it is hardcoded, but only because it is not worth detecting this corner case.
+
+def execute(code, file, mode):
+    exec(code)  # Questionable. Dynamic code
+    exec(compile(code, file, mode))  # Questionable. Dynamic command
+
+exec(hardcoded_command)  # Ok
```

Comment:
RSPEC description does not match implementation, we should clear this `S1523.html` page from what is not supported, like hard coded string literals and variables, and ensure that there's a ticket to improve the rule later to match RSPEC.